### ISSUE TITLE
How to use swiftlint analyze?  And `swiftlint analyze` reports extra warnings beyond `analyzer_rules` configuration.

### ISSUE THREAD

**Issue by 7-Yue** [realm/SwiftLint#6199]
### Problem

In my `.swiftlint.yml` file, I only set the analyzer rule:
```yaml
analyzer_rules:
- explicit_self
```

I run code analysis with the following commands:
```
xcodebuild -workspace XXX.xcworkspace \
                    -scheme XXX \
                    -destination 'id=CF68E389-C815-4918-B133-BE4839938B6F' \
                    clean build | tee xcodebuild-build.log

swiftlint analyze --compiler-log-path xcodebuild-build.log
```
However, the results contain not only explicit_self issues, but also other warnings such as:
```
value was defined but never used
```
These extra warnings seem to come from the Xcode build result itself, but I expected swiftlint analyze to only report issues from analyzer_rules.

### Question
1.	How can I make swiftlint analyze report only issues defined in analyzer_rules?
2.	Are there more examples or best practices for using swiftlint analyze? Documentation is very limited.

**Comment by SimplyDanny** [comment]
You are right. There is not much to say about Analyzer rules. While they are powerful, they are also very slow and there is only a handful of them. What are specifically interested in learning about them?

**Comment by 7-Yue** [comment]
I was expecting to use these rules to guide development habits.
But when I used them,  I found that they didn't work well.
And there are only a few documents about the Analyzer.
So SwiftLint's Analyzer  is not a fully-fledged tool, is it?

**Comment by SimplyDanny** [comment]
It's not an own tool, but a mode to run SwiftLint in. In this mode, only enabled Analyzer rules will be run (all of them are opt-in). The existing one are:

- [capture_variable](https://realm.github.io/SwiftLint/capture_variable.html)
- [explicit_self](https://realm.github.io/SwiftLint/explicit_self.html)
- [typesafe_array_init](https://realm.github.io/SwiftLint/typesafe_array_init.html)
- [unused_declaration](https://realm.github.io/SwiftLint/unused_declaration.html)
- [unused_import](https://realm.github.io/SwiftLint/unused_import.html)

What's more important, Analyzer rules can be much more complex and accurate due to having type information available. But that comes at the cost of runtime and setup. SwiftLint needs to know how to compile a source file being analyzed. That's why we need to provide build log files to the `swiftlint analyze` command. The whole process that then starts is massively slower than the standard rules.
